tests/sles4sap/saptune/mr_test.pm

```diff
@@ -63,18 +73,35 @@ sub setup {
         assert_script_run "sed -ri '/:scripts\\/disk_elevator/s/^/#/' \$(grep -F -rl :scripts/disk_elevator Pattern/)";
         # Skip nr_requests on VM's. Fix bsc#1177888
         assert_script_run 'sed -i "/:scripts\/nr_requests/s/^/#/" Pattern/SLE15/testpattern_*';
+        # Skip tcp_keepalive on public cloud
+        assert_script_run 'sed -i "/:\/proc\/sys\/net\/ipv4\/tcp_keepalive/s/^/#/" Pattern/SLE15/testpattern_*';
+        assert_script_run 'sed -i "/:\/proc\/sys\/net\/ipv4\/tcp_keepalive/s/^/#/" Pattern/SLE12/testpattern_*';
     }
     $self->reboot_wait;
 }
 
 sub run {
-    my ($self) = @_;
+    my ($self, $args) = @_;
 
+    # Preserve args for post_fail_hook
+    $self->{provider} = $args->{my_provider};    # required for cleanup
     $self->setup;
 
     my $test_list = get_required_var("MR_TEST");
     record_info("MR_TEST=$test_list");
-    mr_test_lib::load_mr_tests("$test_list");
+    mr_test_lib::load_mr_tests("$test_list", $args);
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        select_host_console(force => 1);
+        my $args = OpenQA::Test::RunArgs->new();
+        $args->{my_provider} = $self->{provider};
+        autotest::loadtest('tests/publiccloud/ssh_interactive_end.pm', run_args => $args);
```

Comment:
if you need to cleanup just call it directly no need to load whole module just to achieve this

Reply:
Done.